### PR TITLE
PV2-3203 - .NET 6 upgrade

### PR DESCRIPTION
**Pull request by LiamWright**
Note:
AT's removed due to strategy of keeping all AT's in core das-payments-v2 repo after migration.

Pre-release DAS packages will be upgraded to full release when available

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.ServiceFabric/packages.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (David-B-Read) [inline src/SFA.DAS.Payments.RequiredPayments.ServiceFabric/packages.config:1]: Would not expect to see a packages.config file for a .net6 app, is this file required?

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/SFA.DAS.Payments.RequiredPayments.Application.UnitTests.csproj [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+	<PackageReference Include="Autofac" Version="7.0.1" />
+    <PackageReference Include="Autofac.Extras.Moq" Version="6.1.1" />
     <PackageReference Include="AutoFixture.NUnit3" Version="4.11.0" />
     <PackageReference Include="AutoMapper" Version="7.0.1" />
     <PackageReference Include="ESFA.DC.DateTimeProvider" Version="0.1.6-beta0001" />
-    <PackageReference Include="FluentAssertions" Version="5.6.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="2.2.1" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
+    <PackageReference Include="FluentAssertions" Version="5.9.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="7.0.20" />
```
Comment (David-B-Read) [inline src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/SFA.DAS.Payments.RequiredPayments.Application.UnitTests.csproj:14]: This version of EF Core is now out of support. We should target the latest 6.0.x version for .NET6

https://learn.microsoft.com/en-us/ef/core/what-is-new/

File: src/SFA.DAS.Payments.RequiredPayments.Application/SFA.DAS.Payments.RequiredPayments.Application.csproj [hunk 3]
```diff
@@ -15,15 +15,18 @@
...
     <PackageReference Include="AutoMapper" Version="7.0.1" />
+    <PackageReference Include="ESFA.DC.DateTimeProvider" Version="0.1.6-beta0001" />
     <PackageReference Include="ESFA.DC.Logging.Interfaces" Version="0.1.4-beta0001" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="2.2.1" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.2.1" />
-    <PackageReference Include="SFA.DAS.Payments.Application" Version="1.1.0" />
-    <PackageReference Include="SFA.DAS.Payments.Core" Version="1.1.0" />
-    <PackageReference Include="SFA.DAS.Payments.DataLocks.Messages" Version="1.0.0" />
-    <PackageReference Include="SFA.DAS.Payments.EarningEvents.Messages" Version="0.1.123" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.20" />
```
Comment (David-B-Read) [inline src/SFA.DAS.Payments.RequiredPayments.Application/SFA.DAS.Payments.RequiredPayments.Application.csproj:22]: Same comment as above re .NET 7/ EF Core 7

File: src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService.Interfaces/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService.Interfaces.csproj [hunk 4]
```diff
@@ -10,13 +10,13 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ServiceFabric" Version="6.5.664" />
-    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.4.664" />
-    <PackageReference Include="Microsoft.ServiceFabric.Data" Version="3.4.664" />
-    <PackageReference Include="Microsoft.ServiceFabric.Diagnostics.Internal" Version="3.4.664" />
-    <PackageReference Include="Microsoft.ServiceFabric.Services" Version="3.4.664" />
-    <PackageReference Include="SFA.DAS.Payments.DataLocks.Messages" Version="1.0.0" />
-    <PackageReference Include="SFA.DAS.Payments.EarningEvents.Messages" Version="0.1.123" />
+	<PackageReference Include="Microsoft.ServiceFabric" Version="11.0.1000" />
```
Comment (David-B-Read) [inline src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService.Interfaces/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService.Interfaces.csproj:13]: Do we need a DevOps ticket / sub-task to update the ServiceFabric infrastructure to a new version? Can this be done just for the RequiredPayments application or does it need to be done for all of them?

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/SFA.DAS.Payments.RequiredPayments.Application.UnitTests.csproj (hunk 2 above)
Comment (LiamWright) [reply src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/SFA.DAS.Payments.RequiredPayments.Application.UnitTests.csproj]: Bumped down to 6.0.31

File: src/SFA.DAS.Payments.RequiredPayments.Application/SFA.DAS.Payments.RequiredPayments.Application.csproj (hunk 3 above)
Comment (LiamWright) [reply src/SFA.DAS.Payments.RequiredPayments.Application/SFA.DAS.Payments.RequiredPayments.Application.csproj]: Bumped down to 6.0.31

**Review by David-B-Read — DISMISSED** [review]

**Comment by LiamWright** [comment]
PR superseded by https://github.com/SkillsFundingAgency/das-payments-v2-requiredpayments/pull/6
